baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/PreferAssertj.java

```diff
@@ -0,0 +1,406 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.ImmutableSet;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.ImportTree;
+import com.sun.source.tree.MemberSelectTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+import java.util.List;
+import java.util.function.BiConsumer;
+
+/**
+ * {@link PreferAssertj} provides an automated path from legacy test libraries to AssertJ. Our goal is to migrate
+ * existing assertions to AssertJ without losing any information. It's an explicit non-goal to take advantage
+ * of better assertions provided by AssertJ, because those should be implemented in such a way to improve poor
+ * uses of AssertJ as well, which may run after the suggested fixes provided by this checker.
+ *
+ * Remaining work: This doesn't handle Assert.assertThat(value, matcher). We should be able to handle common
+ * cases relatively easily.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "PreferAssertj",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = BugPattern.SeverityLevel.SUGGESTION,
+        summary = "Prefer AssertJ fluent assertions")
+public final class PreferAssertj extends BugChecker implements BugChecker.MethodInvocationTreeMatcher {
+
+    private static final String MESSAGE = "Prefer AssertJ fluent assertions";
```

Comment:
not sure if this message is necessary if it's the same as the BugPattern's message, the fix should inherit that one.

Reply:
setMessage

> Set a custom error message for this {@code Description}. The custom message will be used
> instead of the summary field as the text for the diagnostic message.

Reply:
Oh very nice!